main.go

```diff
@@ -64,6 +64,13 @@ func main() {
 		log.Fatal("usage error")
 	}
 
+	if(len(os.Getenv("AS_USER"))) == 0 {
```

Comment:
I think you have the logic the wrong way around :)

How about:

```
if user := os.Getenv("AS_USER"); user != "" {
  *username = user
}
```
that way you won't repeat the `AS_USER` string.

Reply:
My bad. Updated the commit